cpg-analysis/src/main/kotlin/de/fraunhofer/aisec/cpg/analysis/MultiValueEvaluator.kt

```diff
@@ -179,26 +179,30 @@ class MultiValueEvaluator : ValueEvaluator() {
         }
 
         // We are only interested in expressions
-        val expressions = prevDFG.filterIsInstance<Expression>()
+        // val expressions = prevDFG.filterIsInstance<Expression>()
 
         if (
-            expressions.size == 2 &&
-                expressions.all { e ->
-                    (e.astParent?.astParent as? ForStatement)?.initializerStatement == e ||
-                        (((e.astParent?.astParent as? ForStatement)?.initializerStatement
-                                as? DeclarationStatement)
-                            ?.singleDeclaration != null &&
-                            ((e.astParent?.astParent as? ForStatement)?.initializerStatement
-                                    as? DeclarationStatement)
-                                ?.singleDeclaration == e.astParent) ||
-                        (e.astParent as? ForStatement)?.iterationStatement == e
+            prevDFG.size == 2 &&
```

Comment:
The next few lines are really hard to understand. Maybe refactor them a little bit?